docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+
+Though both these types are equally important, for this guide, we'll focus on client-side validation only.
+
+Client-side validation is further categorized as:
+
+- **Built-in**: Uses HTML-based attributes like `required`, `type`, `minLength`, `maxLength`, `pattern` etc.
+- **JavaScript-based**: Validation that's coded with JavaScript.
+
+### Built-in Form Validation Using `required`, `type`, `minLength`, `maxLength`
+
+- `required`: Specifies which fields must be filled before submitting the form.
+- `type`: Tells whether the data should be a number, email id, text string, etc.
+- `minLength`: Decides the minimum length for the text data string.
+- `maxLength`: Decides the maximum length for the text data string.
+
+Let's modify the form w.r.t. to these attributes:
+
+```html
+<!-- HTML Form with Built-in Validation -->
+<form action="/send-data-here" method="post">
+  <label for="roll">Roll Number</label>
+  <input
+    type="text"
+    id="roll"
+    name="roll"
+    required
+    minlength="10"
+    maxlength="20"
+  />
+  <label for="name">Name:</label>
+  <input type="text" id="name" name="name" required />
+  <button type="submit">Submit</button>
+</form>
+```
+
+With these validation checks in place, when a user tries to submit an empty field for Last Name, it gives an error that pops right in the form field. Similarly, a roll number can only be entered if it's 10-20 characters long.
+
+![form validation](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009088/nextjs/guides/building-forms/form-validation.jpg)
+
+### JavaScript-based Form Validation
+
+Form Validation is important to ensure that a user has submitted appropriate data. JavaScript offers an additional level of validation along with HTML native form attributes on the client side. Developers generally prefer validating form data through JavaScript because its data processing is faster when compared to server-side validation.
+
+Let's look at a simple example:
+
+```html
+<form onsubmit="return validateFormWithJS()">
+  <label for="name">Name:</label>
+  <input type="text" name="name" id="name" />
+
+  <label for="rollNumber">Roll Number:</label>
+  <input type="text" name="rollNumber" id="rollNumber" />
+
+  <button type="submit">Submit</button>
+</form>
+
+<script>
+  function validateFormWithJS() {
+    const name = document.querySelector('#name').value
+    const rollNumber = document.querySelector('#rollNumber').value
+
+    if (!name) {
+      alert('Please enter your name.')
+      return false
+    }
+
+    if (rollNumber.length < 3) {
+      alert('Roll Number should be at least 3 digits long.')
+      return false
+    }
+  }
+</script>
+```
+
+The HTML [script](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) tag is used to embed any client-side JavaScript. It can either contain inline scripting statements (as shown in the example above) or point to an external script file via the src attribute.
```

Comment:
```suggestion
The HTML [script](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) tag is used to embed any client-side JavaScript. It can either contain inline scripting statements (as shown in the example above) or point to an external script file via the `src` attribute.
```